### ISSUE TITLE
Support to o1-preview

### ISSUE THREAD

**Issue by marcelcastrobr** [langchain-ai/report-mAIstro#7]
**Issue**: Error when using o1-preview model from AzureChatOpenAI.

**Error in Langgraph**: BadRequestError('Error code: 400 - {\'error\': {\'message\': "Unsupported value: \'messages[0].role\' does not support \'system\' with this model.", \'type\': \'invalid_request_error\', \'param\': \'messages[0].role\', \'code\': \'unsupported_value\'}}')

Error log console:   
`File "/home/marcec/github/report-mAIstro/report_masitro.py", line 429, in generate_report_plan
    results = structured_llm.invoke([SystemMessage(content=system_instructions_query)]+[HumanMessage(content="Generate search queries that will help with planning the sections of the report.")])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_core/runnables/base.py", line 3022, in invoke
    input = context.run(step.invoke, input, config, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_core/runnables/base.py", line 5354, in invoke
    return self.bound.invoke(
           ^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_core/language_models/chat_models.py", line 286, in invoke
    self.generate_prompt(
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_core/language_models/chat_models.py", line 786, in generate_prompt
    return self.generate(prompt_messages, stop=stop, callbacks=callbacks, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_core/language_models/chat_models.py", line 643, in generate
    raise e
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_core/language_models/chat_models.py", line 633, in generate
    self._generate_with_cache(
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_core/language_models/chat_models.py", line 839, in _generate_with_cache
    for chunk in self._stream(messages, stop=stop, **kwargs):
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/langchain_openai-0.2.11-py3.12.egg/langchain_openai/chat_models/base.py", line 640, in _stream
    response = self.client.create(**payload)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/openai-1.57.0-py3.12.egg/openai/_utils/_utils.py", line 275, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/openai-1.57.0-py3.12.egg/openai/resources/chat/completions.py", line 829, in create
    return self._post(
           ^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/openai-1.57.0-py3.12.egg/openai/_base_client.py", line 1280, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/openai-1.57.0-py3.12.egg/openai/_base_client.py", line 957, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/openai-1.57.0-py3.12.egg/openai/_base_client.py", line 1061, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'error': {'message': "Unsupported value: 'messages[0].role' does not support 'system' with this model.", 'type': 'invalid_request_error', 'param': 'messages[0].role', 'code': 'unsupported_value'}}
During task with name 'generate_report_plan' and id '2419dcdd-e2c6-26a0-e8cb-9816b8ab7c1e'`

**Comment by marcelcastrobr** [comment]
Changed SystemMessage by HumanMessage.
But next issue is with the tools.
Error: 'tools' is not supported with this model.

`  File "/home/marcec/miniconda3/envs/langraph/lib/python3.12/site-packages/openai-1.57.0-py3.12.egg/openai/_base_client.py", line 1061, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'error': {'message': "Unsupported parameter: 'tools' is not supported with this model.", 'type': 'invalid_request_error', 'param': 'tools', 'code': 'unsupported_parameter'}}`

**Comment by marcelcastrobr** [comment]
ref. O1 limitations: https://platform.openai.com/docs/guides/reasoning/beta-limitations 
🥺 

_Beta Limitations
During the beta phase, many chat completion API parameters are not yet available. Most notably:
Modalities: text only, images are not supported.
Message types: user and assistant messages only, system messages are not supported.
Tools: tools, function calling, and response format parameters are not supported.
Logprobs: not supported.
Other: temperature and top_p are fixed at 1, while presence_penalty and frequency_penalty are fixed at 0.
Assistants and Batch: these models are not supported in the Assistants API or Batch API. _

**Comment by cmosguy** [comment]
@marcelcastrobr  yes exactly, so what you have to do is write a plan that describes the functions that should be executed, then you pass that plan onto a gpt-4o model.  Here is the pattern described in the course here: https://learn.deeplearning.ai/courses/reasoning-with-o1/lesson/4/planning-with-o1

**Comment by rlancemartin** [comment]
Added to the planner! And set up w/ o3-mini :)
